MAGECLOUD-1897: Change .htaccess and .gitkeep in templates (#405)
magento/magento-cloud: m2-hotfixes/README.md

# Add custom patches

You can add custom patch files to this directory. All custom patch filenames must have the `.patch` extension, for example `<patch_name>.patch`.
After all Magento patches are applied, custom patches are applied in alphabetical order from the root of the Magento project as shown in the following example:

    ```
    diff -Naur a/vendor/package-name/ClassName.php b/vendor/package-name/ClassName.php
    --- a/vendor/package-name/ClassName.php
    +++ b/vendor/package-name/ClassName.php
    ...
    ```
    

> You can use the following command to test the patch process–applying all Magento patches and your custom patches  locally before you build and deploy to a Cloud environment.

Reviewer: Suggested change:

```
You can use the following command to apply all Magento patches and your custom patches locally.
```

The other wording is awkward, and honestly I don't think you need any more specific details than that.

Author: done.

> You can use the following command to test the patch process–applying all Magento patches and your custom patches  locally before you build and deploy to a Cloud environment.

Reviewer 2: ```suggestion
You can use the following command to apply all Magento patches and your custom patches locally.
```

# Add custom patches

You can add custom patch files to this directory. All custom patch filenames must have the `.patch` extension, for example `<patch_name>.patch`.
After all Magento patches are applied, custom patches are applied in alphabetical order from the root of the Magento project as shown in the following example:

    ```
    diff -Naur a/vendor/package-name/ClassName.php b/vendor/package-name/ClassName.php
    --- a/vendor/package-name/ClassName.php
    +++ b/vendor/package-name/ClassName.php
    ...
    ```
    
You can use the following command to test the patch process–applying all Magento patches and your custom patches  locally before you build and deploy to a Cloud environment.
``` 

> ./vendor/bin/ece-tools patch

Reviewer 3: This is deprecated in `2002.1`

Author: @shiftedreality what will be instead?

Author: and if we plan to deliver these changes before 2002.1 then better not to mention this not released version

Reviewer 3: Direct call to patches `./vendor/bin/ece-patches apply`

Author: @YPyltiai will we deliver this earlier that minor version of ece-tools? or wait till its release and add command as for 2002.0 as for 2002.1?

# Add custom patches

> You can add custom patch files to this directory. Each patch file must use the `.patch` extension, for example the `<patch_name>.patch`.

Reviewer 2: ```suggestion
You can add custom patch files to this directory. You must use the `.patch` extension in the custom patch file name, for example `<patch_name>.patch`.
```

> There are rules about patches in this directory:

Reviewer 2: After all Magento patches are applied, custom patches are applied in alphabetical order from the root of the Magento project as shown in the following example:

>  - patch file should have extension `.patch`

Reviewer 2: ```suggestion
```

>  - patches are applied in alphabetical sequence

Reviewer 2: ```suggestion
 - patches are applied in alphabetical sequence
```
```suggestion
```

>  - patches are applied from the root of Magento project. E.g.

Reviewer 2: ```suggestion
```

 
    ```
    diff -Naur a/vendor/package-name/ClassName.php b/vendor/package-name/ClassName.php
    --- a/vendor/package-name/ClassName.php
    +++ b/vendor/package-name/ClassName.php
    ...
    ```
    

> To test applying all patches (your custom patches and cloud patches) which is made during the deployment, you can run next command locally:

Reviewer 2: ```suggestion
You can use the following command to test the patch process–applying all Magento patches and your custom patches  locally before you build and deploy to a Cloud environment.
```
